src/lib.rs

```diff
@@ -150,10 +156,21 @@ impl OutputArg {
         self.output.as_deref()
     }
 
-    pub fn output_writer(&self) -> Result<Box<dyn Write>> {
+    pub fn output_writer(&self, mut color: ColorChoice) -> Result<Box<dyn WriteColor>> {
+        if color == ColorChoice::Auto && !atty::is(atty::Stream::Stdout) {
+            color = ColorChoice::Never;
+        }
```

Comment:
For this logic `atty` only need be consulted if `self.output` is `None`, otherwise in "auto" mode with a file as output there's no need to consult `atty`.